src/schemes/local.ts

```diff
@@ -36,39 +40,61 @@ const DEFAULTS: SchemeOptions = {
 }
 
 export default class LocalScheme extends BaseScheme<typeof DEFAULTS> {
-  requestHandler: RequestHandler
+  public token: Token
+  public requestHandler: RequestHandler
 
   constructor ($auth, options, ...defaults) {
     super($auth, options, ...defaults, DEFAULTS)
 
+    // Initialize Token instance
+    this.token = new Token(this, this.$auth.$storage)
+
     // Initialize Request Interceptor
-    this.requestHandler = new RequestHandler(this.$auth)
+    this.requestHandler = new RequestHandler(this, this.$auth.ctx.$axios)
   }
 
   _updateTokens (response) {
     if (this.options.token.required) {
       const token = getResponseProp(response, this.options.token.property)
-      this.$auth.token.set(token)
+      this.token.set(token)
     }
   }
 
-  _checkStatus () {
-    if (this.options.token.required) {
-      // Sync token
-      this.$auth.token.sync()
+  check (checkStatus = false, tokenCallback?, _refreshTokenCallback?) {
+    // Sync token
+    const token = this.token.sync()
 
-      // Get token status
-      const tokenStatus = this.$auth.token.status()
+    // Token is required but not available
+    if (this.options.token.required && !token) {
+      return false
+    }
 
-      // Token is expired. Force reset.
-      if (tokenStatus.expired()) {
-        this.$auth.reset()
+    // Check status wasn't enabled, let it pass
+    if (!checkStatus) {
+      return true
+    }
+
+    // Get status
+    const tokenStatus = this.token.status()
+
+    // Token has expired. Attempt `tokenCallback`
+    if (tokenStatus.expired()) {
+      if (typeof tokenCallback === 'function') {
+        return tokenCallback(false) || false
```

Comment:
We can instead, return either false or false or an object like `{ tokenExpired: true }` so check is still not-falsy but can provide better meta.

Reply:
Improved in commit 4d014e13